cpp/src/parquet/statistics.h

```diff
@@ -291,6 +291,9 @@ class TypedStatistics : public Statistics {
   /// arrow::BinaryArray
   virtual void Update(const ::arrow::Array& values) = 0;
 
+  virtual void UpdateArrowDictionary(const ::arrow::Array& indices,
```

Comment:
can this be moved to the caller instead.  It would be nice to try to avoid more coupling of Arrow with the default parquet library.

Reply:
I can move the "compute referenced values" logic to the caller so that statistics.cc doesn't depend on ::arrow::compute but I think I still need a second Update method here because the null count comes from the indices array and the min/max from the values array.

Alternatively, I could change the existing Update method to take an optional null_count parameter (where -1 flags to use the old behavior and grab from the array).  I think I'll do this latter approach.

Reply:
I think making update take an optional null_count parameter actually makes the most sense.  Not sure if you are subscribed to dev@parquet but the null_count statistics are incorrect for repeated fields https://issues.apache.org/jira/browse/PARQUET-2067 and I think the fix will involve passing along the null count as well.

Reply:
I ended up just exposing `IncrementNullCount` and `IncrementNumValues` and then added `bool update_counts` to `Update`.